internal/juju/jaas_test.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the Apache License, Version 2.0, see LICENCE file for details.
+
+package juju
+
+import (
+	"testing"
+
+	"github.com/canonical/jimm-go-sdk/v3/api/params"
+	"github.com/juju/juju/api"
+	"github.com/stretchr/testify/suite"
+	"go.uber.org/mock/gomock"
+)
+
+type JaasSuite struct {
+	suite.Suite
+	JujuSuite
+
+	mockJaasClient *MockJaasAPIClient
+}
+
+func (s *JaasSuite) SetupTest() {}
+
+func (s *JaasSuite) setupMocks(t *testing.T) *gomock.Controller {
+	ctlr := s.JujuSuite.setupMocks(t, nil)
+	s.mockJaasClient = NewMockJaasAPIClient(ctlr)
+
+	return ctlr
+}
+
+func (s *JaasSuite) getJaasClient() jaasClient {
+	return jaasClient{
+		SharedClient: s.JujuSuite.mockSharedClient,
+		getJaasApiClient: func(connection api.Connection) JaasAPIClient {
+			return s.mockJaasClient
+		},
+	}
+}
+
+func (s *JaasSuite) TestAddRelations() {
+	ctlr := s.setupMocks(s.T())
+	defer ctlr.Finish()
+
+	tuples := []params.RelationshipTuple{
```

Comment:
we don't test for cases where AddRelations is called with no tuples

Reply:
In this mock test that wouldn't change anything except unless we added a check for an empty slice.

Reply:
Added test for this.